jspdf.js

```diff
@@ -2089,21 +2450,31 @@ var jsPDF = (function(global) {
      * @methodOf jsPDF#
      * @name setTextColor
      */
-    API.setTextColor = function(r, g, b) {
-      if ((typeof r === 'string') && /^#[0-9A-Fa-f]{6}$/.test(r)) {
-        var hex = parseInt(r.substr(1), 16);
-        r = (hex >> 16) & 255;
-        g = (hex >> 8) & 255;
-        b = (hex & 255);
-      }
+    API.setTextColor = function (r, g, b, flags) {
+      var objColor;
+      var r1, g1, b1;
+
+      function colorcalc(r, g, b) {
+        var Color;
+        if ((typeof r === 'string') && /^#[0-9A-Fa-f]{6}$/.test(r)) {
+          var hex = parseInt(r.substr(1), 16);
+          r = (hex >> 16) & 255;
+          g = (hex >> 8) & 255;
+          b = (hex & 255);
+        }
 
-      if ((r === 0 && g === 0 && b === 0) || (typeof g === 'undefined')) {
-        textColor = f3(r / 255) + ' g';
-      } else {
-        textColor = [f3(r / 255), f3(g / 255), f3(b / 255), 'rg'].join(
-          ' ');
+        if ((r === 0 && g === 0 && b === 0) || (typeof g === 'undefined')) {
+          return Color = f3(r / 255) + ' g';
+        } else {
+          return Color = [f3(r / 255), f3(g / 255), f3(b / 255), 'rg'].join(' ');
+        }
+      }
+      if (flags) { //If the color is multi-attribute
```

Comment:
Why does your modified method returns the color variable and sometimes the this-Object? 
Imho critical modification which makes the method unpredictable.

Reply:
It was made because it seemed necessary when setting multi attributes on one line.
I have created it with a different API function, is that better?